Roomモデルを作成する

uchihiro04 opened the pull request:
## 概要 
Roomモデルを作成しました。またバリデーションを設定しました。

## スクリーンショット
* Roomsテーブル
<img width="700" alt="スクリーンショット 2023-06-22 15 05 21" src="https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/8bc064a4-bb66-490e-a0d8-4a8fe85e70a6">

* レコードの作成
<img width="700" alt="スクリーンショット 2023-06-22 15 06 20" src="https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/c4d5fa0b-ae1e-481c-811f-f07d7141ef19">

* レコードの作成（name, uuidがnilのとき）
<img width="700" alt="スクリーンショット 2023-06-22 15 06 46" src="https://github.com/uchihiro04/house-viewing-score-log/assets/77523896/b946b437-6541-4520-85ab-4a2a805059ea">

## 関連Issue
- #76 

## 備考
### house_viewing_idについて
当初はhouse_viewing_idにユニーク制約がついていましたが、同じhouse_viewing_idを持ったRoomデータを複数作成できなくなるため、ユニーク制約を取り外しています。以下のリンクでも同じ修正を行いました。
- #5 
- #76 

Roomsテーブルは以下の形を想定しています。
| カラム名 | データ型 | PK | FK | UQ | NN |
| --- | --- | --- | --- | --- | --- |
| id | integer | ○ |  | ○ | ○ | 
| house_viewing_id | integer |  | ○ |  | ○ | 
| name | string |  |  |  | ○ | 
| created_at | datetime |  |  |  | ○ | 
| updated_at | datetime |  |  |  | ○ | 

## コミットについて
実装で必要となったため、[HouseViewingモデルの作成](https://github.com/uchihiro04/house-viewing-score-log/commit/c2300b55208f7b1c473550be25b1ee56aa944dd8)から[マイグレーションの実行](https://github.com/uchihiro04/house-viewing-score-log/commit/d7ef52ede1cb187a6fa357c9b795ad3d225513d0)まではチェリーピックをしています。

uchihiro04 committed:
Roomモデルを作成

uchihiro04 committed:
マイグレーションを実行

house_viewing_idカラムを外部キーにしました。また、nameおよびhouse_viewing_idカラムにNOT NULL制約を追加しました。

uchihiro04 committed:
バリデーションを追加

uchihiro04 committed:
マジックコメントを追加

rubocopからの指摘を受け、マジックコメントを追加しました。

takuminasu commented on app/models/room.rb:
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class Room < ApplicationRecord
+  validates :house_viewing_id, presence: true
```
validatesよりbelongs_toの方がいいと思います。

uchihiro04 replied:
house_viewing_idのバリデーションを削除しました。該当するコミットは[3f92e16](https://github.com/uchihiro04/house-viewing-score-log/pull/98/commits/3f92e16f13485424d6334e7bacb2bd14c40b4c6f)です。
belongs_to関連付けに関しましては、[HouseViewingデータ作成時にRoomデータを10個作成する](https://github.com/uchihiro04/house-viewing-score-log/pull/103)の[22f6d73](https://github.com/uchihiro04/house-viewing-score-log/pull/103/commits/22f6d738b02868225ec1208e2676d86fb5af4b61)で実装しています。

uchihiro04 committed:
house_viewing_idに関するバリデーションを削除

belongs_toを使用した場合、外部キーの存在について二重でチェックが走ってしまうため、house_viewing_idに関するバリデーションを削除しました。

takuminasu reviewed (approved):
LGTM
